bundles/specmate-emfrest/src/com/specmate/emfrest/internal/AuthentificationFilter.java

```diff
@@ -1,70 +1,75 @@
 package com.specmate.emfrest.internal;
 
 import java.io.IOException;
-import java.util.List;
-import java.util.StringTokenizer;
 
 import javax.inject.Inject;
 import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.container.ContainerRequestFilter;
-import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
 
-import org.apache.commons.lang3.tuple.ImmutablePair;
-import org.apache.commons.lang3.tuple.Pair;
-import org.glassfish.jersey.internal.util.Base64;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.log.LogService;
 
 import com.specmate.auth.api.IAuthentificationService;
+import com.specmate.common.SpecmateException;
+import com.specmate.emfrest.authentication.Login;
 
+@Secured
+@Provider
+@Component(immediate=true, service=ContainerRequestFilter.class)
 public class AuthentificationFilter implements ContainerRequestFilter {
-
-	private static final String AUTHORIZATION_PROPERTY = "Authorization";
-	private static final String AUTHENTICATION_SCHEME = "Basic";
-	private static final Response ACCESS_DENIED = Response.status(Response.Status.UNAUTHORIZED)
-			.entity("You cannot access this resource").build();
-	private static final Response ACCESS_FORBIDDEN = Response.status(Response.Status.FORBIDDEN)
-			.entity("Access blocked for all users !!").build();
-
-	@Inject
-	private IAuthentificationService authenticationService;
-
+	private static final String REALM = "specmate";
+   
+    @Inject
+    IAuthentificationService authService;
+    
+    @Inject
+    LogService logService;
+    
 	@Override
 	public void filter(ContainerRequestContext requestContext) throws IOException {
-
-		// Get request headers
-		final MultivaluedMap<String, String> headers = requestContext.getHeaders();
-
-		final List<String> authorization = headers.get(AUTHORIZATION_PROPERTY);
-		// If no authorization information present; block access
-		// if (authorization == null || authorization.isEmpty()) {
-		// requestContext.abortWith(ACCESS_DENIED);
-		// return;
-		// }
-		//
-		// Pair<String, String> userNameAndPassword =
-		// extractUserAndPassword(authorization.get(0));
-
-		String url = requestContext.getUriInfo().getMatchedURIs(true).get(0);
-
-		// if (!authenticationService.authenticate("admin", "admin", url)) {
-		// requestContext.abortWith(ACCESS_DENIED);
-		// return;
-		// }
+		// Certain URIs should not be secured, e.g. login.
+		if (isNotSecured(requestContext)) {
+			return;
+		}
+		
+		// Get the Authorization header from the request
+        String authorizationHeader =
+                requestContext.getHeaderString(HttpHeaders.WWW_AUTHENTICATE);
+
+        // Validate the Authorization header
+        if (authorizationHeader == null) {
+        	logService.log(LogService.LOG_INFO, "Invalid authorization header: " + authorizationHeader);

```

Comment:
For debugging reasons: Could you add, how the header should actually look like?

Reply:
Currently, it is simple: WWW-Authenticate: token
This will change with the Digest access authentication, mentioned above.

Reply:
Sorry, made a mistake here. The correct header is: 
`Authorization: token`
I fixed it in commit 20e55341639b22c9be06a80521d4f4e3c27bad22.